src/Headless/Avalonia.Headless/HeadlessWindowImpl.cs

```diff
@@ -413,5 +413,7 @@ public void SetFrameThemeVariant(PlatformThemeVariant themeVariant)
         {
             
         }
+
+        public void GetWindowsZOrder(Span<Window> windows, Span<long> zOrder) => throw new NotSupportedException();
```

Comment:
Headless platform should try to mimic real platform as much as possible, and not throw exceptions.
For example, we can keep static counter and increment it in HeadlessWindow.Activate(), saving locally per window. And use incremented index in GetWindowsZOrder. A simple solution, which can be then used in tests by somebody.